rts/motoko-rts-tests/src/gc.rs

```diff
@@ -67,17 +95,6 @@ fn test_gc(
 ) {
     let heap = MotokoHeap::new(refs, roots, closure_table, gc);
 
-    // Check `check_dynamic_heap` sanity
-    check_dynamic_heap(
-        refs,
-        roots,
-        closure_table,
-        &**heap.heap(),
-        heap.heap_base_offset(),
-        heap.heap_ptr_offset(),
-        heap.closure_table_ptr_offset(),
-    );
```

Comment:
Same as previous PR, this is removed as `check_dynamic_heap` rejects garbage, but we want to test heaps with garbage as well. So currently `check_dynamic_heap` is only called after a GC, not before.